### PR TITLE
Fix empty fields causing AR reader to fail

### LINKED ISSUE #129: `Ar.Next()` error: `strconv.Atoi: parsing \"\": invalid syntax`
When attempting to read `*.a` files, an error is occasionally encountered:
```
strconv.Atoi: parsing \"\": invalid syntax
```

The source of this is `Ar.Read > parseArEntry > toDecimal`. It seems that one of the values is empty, resulting in an error.
1. https://github.com/paultag/go-debian/blob/246d63d4a8289011dff5db890058a7d9403f785a/deb/ar.go#L89
3. https://github.com/paultag/go-debian/blob/246d63d4a8289011dff5db890058a7d9403f785a/deb/ar.go#L153
4. https://github.com/paultag/go-debian/blob/246d63d4a8289011dff5db890058a7d9403f785a/deb/ar.go#L107

This can be reproduced with the following file: https://github.com/openshift/ci-operator-prowgen/blob/ee31ec067b4670222de32ceecd01184b4eb52db3/vendor/golang.org/x/tools/go/gccgoexportdata/testdata/long.a

### PR DESCRIPTION
**Pull request by rgmz**
This fixes #129.

### CODE REVIEW

File: deb/ar.go [hunk 1]
```diff
@@ -144,17 +147,22 @@ func parseArEntry(line []byte) (*ArEntry, error) {
...
-	for target, value := range map[*int64][]byte{
-		&entry.Timestamp: line[16:28],
-		&entry.OwnerID:   line[28:34],
-		&entry.GroupID:   line[34:40],
-		&entry.Size:      line[48:58],
+	for target, value := range map[entryField][]byte{
+		entryField{"Timestamp", &entry.Timestamp}: line[16:28],
+		entryField{"OwnerID", &entry.OwnerID}:     line[28:34],
+		entryField{"GroupID", &entry.GroupID}:     line[34:40],
+		entryField{"Size", &entry.Size}:           line[48:58],
```
Comment (rgmz) [inline deb/ar.go:154]: I added this struct because it makes errors clearer and debugging easier. If this isn't desirable, I can remove it.

**Comment by rgmz** [comment]
Hey @paultag, I was curious if you had any insights into this. 

It seems that ar files can contain special entries (`/` and `//` meaning symbol table and extended filename table respectively[^1]), hence why some fields are empty. I'm not sure if my proposed fix is sufficient, or if additional pre-processing is desirable.

e.g., `ar x long.a`[^2] results in a single file named `name-longer-than-16-bytes.gox`. This seems to differ from this library's current behaviour.
```sh
$ cat long.a
!<arch>
/               1478713642  0     0     0       4         `
//                                              32        `
name-longer-than-16-bytes.gox/

/0              1478713628  10556 5000  100664  577       `
ELF>�@@v1;
package foo;
pkgpath blah;
priority 1;
func F ();
checksum F9F6C8FE0C9E3CDD5A09D2996847B4E9992D8118;
.symtab.strtab.shstrtab.go_exporti�&0   @
```

[^1]: https://en.wikipedia.org/wiki/Ar_(Unix)#System_V_(or_GNU)_variant
[^2]: https://github.com/openshift/ci-operator-prowgen/blob/ee31ec067b4670222de32ceecd01184b4eb52db3/vendor/golang.org/x/tools/go/gccgoexportdata/testdata/long.a

**Comment by paultag** [comment]
@rgmz I don't think the case you've outlined is something we handle particularly well today; and I think what you have is sensible. I'm 100% fine landing this and fixing any fallout as we find it, whatever this breaks on was broken previously, so it's low stress. Thanks for your changes!

**Comment by rgmz** [comment]
> I'm 100% fine landing this and fixing any fallout as we find it, whatever this breaks on was broken previously, so it's low stress.

Sounds like a plan! Are you able to push a new tag so we can test this and #110?

**Comment by paultag** [comment]
`v0.17.0` pushed, thx for the reminder!

**Pull request merged**